src/app/shared/services/records/record.service.ts

```diff
@@ -154,4 +154,21 @@ export class RecordService extends GenericListService {
     data.set('content', feedbackData);
     return this.http.post(`${environment.API_URL}/campus-event-feedbacks/`, data);
   }
+
+  updateRecordStatus(record: Record, isApproved: boolean, isDepartmentAdmin: boolean, isSchoolAdmin: boolean) {
+    const data = {
+      id: record.id,
+      user: record.user,
+      contents: [],
+      attachments: [],
+      feedback: null,
+      is_approved: isApproved
+    };
+    if ((isDepartmentAdmin) && (!isSchoolAdmin)) {
```

Comment:
As this service is used only by school admin or department admin, the ifs can be rewrite to:
```typescript
if (isDepartmentAdmin) {
  // Post to department admin review endpoint.
  return xxx;
}
// Post to school admin review endpoint.
return yyy;
```